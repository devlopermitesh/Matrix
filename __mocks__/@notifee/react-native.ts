
module.exports = {
  __esModule: true,
  default: {
    createChannel: jest.fn(),
    displayNotification: jest.fn(),
    createTriggerNotification: jest.fn(),
    cancelNotification: jest.fn(),
    cancelAllNotifications: jest.fn(),
    getInitialNotification: jest.fn(() => Promise.resolve(null)),
    onForegroundEvent: jest.fn(),
    onBackgroundEvent: jest.fn(),
    requestPermission: jest.fn(() => Promise.resolve({ authorizationStatus: 1 })),
  },
  AndroidImportance: {
    HIGH: 4,
    DEFAULT: 3,
    LOW: 2,
    MIN: 1,
  },
  AndroidStyle: {
    BIGTEXT: 1,
    BIGPICTURE: 2,
    INBOX: 3,
    MESSAGING: 4,
  },
  TriggerType: {
    TIMESTAMP: 0,
    INTERVAL: 1,
  },
  TimeUnit: {
    SECONDS: 0,
    MINUTES: 1,
    HOURS: 2,
    DAYS: 3,
  },
  RepeatFrequency: {
    NONE: -1,
    HOURLY: 0,
    DAILY: 1,
    WEEKLY: 2,
  },
  AndroidAction: {},
  IntervalTrigger: {},
  TimestampTrigger: {},
};