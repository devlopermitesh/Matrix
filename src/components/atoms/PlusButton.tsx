import { Image, StyleSheet, TouchableOpacity, ViewStyle } from "react-native";
import Icon from "./Icon";
const PlusButton = ({style,onPress}:{style:ViewStyle,onPress:()=>void}) => {
  return (
    <TouchableOpacity testID="plus-button" style={[styles.buttonContainer,style]} onPress={onPress}>
      <Image source={require("../../asset/Images/plusIcon.png")} style={{width:"100%",height:'100%',objectFit:"cover"}}/>
    </TouchableOpacity>
  );
};

export default PlusButton;

const styles = StyleSheet.create({
  buttonContainer: {
    width:60,
    height:60,
    backgroundColor: "#1591EA",
    borderRadius: 50, // circle
    borderWidth: 1,
    borderColor: "#fff",
    justifyContent: "center",
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 4,
    elevation: 5, // for Android shadow
  },
});
