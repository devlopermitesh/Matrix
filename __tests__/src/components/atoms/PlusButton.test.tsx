import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import PlusButton from "../../../../src/components/atoms/PlusButton";

describe("PlusButton Component", () => {
  it("renders correctly with default styles", () => {
    const { getByTestId } = render(<PlusButton onPress={() => {}} style={{}} />);
    const button = getByTestId("plus-button");
    expect(button).toBeTruthy();
  });

  it("calls onPress when pressed", () => {
    const mockPress = jest.fn();
    const { getByTestId } = render(<PlusButton onPress={mockPress} style={{}} />);
    const button = getByTestId("plus-button");
    fireEvent.press(button);
    expect(mockPress).toHaveBeenCalledTimes(1);
  });

  it("applies custom style when passed", () => {
    const customStyle = { backgroundColor: "red" };
    const { getByTestId } = render(<PlusButton onPress={() => {}} style={customStyle} />);
    const button = getByTestId("plus-button");
  expect(button.props.style).toEqual((expect.objectContaining(customStyle))
  );


  });
});
