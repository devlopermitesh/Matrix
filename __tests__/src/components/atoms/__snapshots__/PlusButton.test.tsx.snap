// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusButton Component applies custom style when passed 1`] = `
{
  "alignItems": "center",
  "backgroundColor": "red",
  "borderColor": "#fff",
  "borderRadius": 50,
  "borderWidth": 1,
  "elevation": 5,
  "height": 60,
  "justifyContent": "center",
  "opacity": 1,
  "shadowColor": "#000",
  "shadowOffset": {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 4,
  "width": 60,
}
`;
